src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -15249,5 +15261,106 @@ public static void Main()
 True
 null");
         }
+
+        [Fact, WorkItem(10459, "https://github.com/dotnet/roslyn/issues/10459")]
+        public void Typeswitch_01()
+        {
+            var source =
+@"
+using System;
+public class X
+{
+    public static void Main(string[] args)
+    {
+        switch (args.GetType())
+        {
+            case typeof(string):
+                Console.WriteLine(""string"");
+                break;
+            case typeof(string[]):
+                Console.WriteLine(""string[]"");
+                break;
+            case null:
+                Console.WriteLine(""null"");
+                break;
+            default:
+                Console.WriteLine(""default"");
+                break;
+        }
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: patternParseOptions);
+            compilation.VerifyDiagnostics(
+                // (9,18): error CS0150: A constant value is expected
+                //             case typeof(string):
+                Diagnostic(ErrorCode.ERR_ConstantExpected, "typeof(string)").WithLocation(9, 18),
+                // (12,18): error CS0150: A constant value is expected
+                //             case typeof(string[]):
+                Diagnostic(ErrorCode.ERR_ConstantExpected, "typeof(string[])").WithLocation(12, 18)
+                );
+            // If we decide to support switching on System.Type, the following line would be appropriate.
+            // CompileAndVerify(compilation, expectedOutput: @"string[]");
```

Comment:
Our policy is to not submit commented out code. If you want to keep it (the commented out code) because this is a prototype and we would come back to it. Please mark it with a PROTOTYPE marker so that we wouldn't forget to come back to it and uncomment or delete it.